internal/konnect/node_agent.go

```diff
@@ -71,16 +105,36 @@ func (a *NodeAgent) clearOutdatedNodes() error {
 	return nil
 }
 
+func (a *NodeAgent) subscribeConfigStatus(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			err := ctx.Err()
+			a.Logger.Info("subscribe loop stopped", "message", err.Error())
+			return
+		case a.configStatus = <-a.configStatusSubscriber.SubscribeConfigStatus():
```

Comment:
`SubscribeConfigStatus()` returns a channel. Can'e we just store the returned value (channel) in scope of this function an reuse it?

Reply:
More importantly: `updateNode()` is called on a ticker in `updateNodeLoop()`, whereas `subscribeConfigStatus()` here, receives from a channel (returned by `a.configStatusSubscriber.SubscribeConfigStatus()`) that occurs asynchronously when a config is synced.

Isn't that a potential data race? If you set a shorter data plane sync period and `a.refreshPeriod` wouldn't you get a data race? 🤔

Reply:
Used `atomic` to store/load it.